### PR TITLE
Embedded Workflows on OCP docker custom pull secret

### PR DESCRIPTION
**Pull request by aileenjackson**
@Fryguy I've added new content as requested in https://github.ibm.com/katamari/dev-issue-tracking/issues/59552. Can you confirm it's phrased as you would like and in the right place etc? Thanks Aileen

### CODE REVIEW

File: managing_providers/_topics/embedded_workflows.md [hunk 1]
```diff
@@ -280,6 +280,21 @@ Workflows must be authored in Amazon State Languages (ASL) format. As part of au
   }
 }
 ```
+#### Running an Embedded Workflow on the Openshift Container Platform (OCP) 
+
+If the user is running an embedded workflow on OCP, as opposed to appliances, and is using a docker repository requiring a custom pull secret for their container images, the following steps are necessary:
+
+1. create a docker pull secret for the repository in question in the same namespace as IM, for example:
```
Comment (Fryguy) [inline managing_providers/_topics/embedded_workflows.md:287]: "IM" is a downstream specific name, so we should use a conref here.  Also, minor, but this should probably be a full sentence.

```suggestion
1. Create a docker pull secret for the repository in question in the same namespace as {{ site.data.product.title_short }}, for example:
```

File: managing_providers/_topics/embedded_workflows.md [hunk 2]
```diff
@@ -280,6 +280,21 @@ Workflows must be authored in Amazon State Languages (ASL) format. As part of au
...
+
+    ```
+    oc create secret docker-registry <pull-secret-name> --docker-server=<docker-server> --docker-username=<docker-username> --docker-password=<docker-password> 
+    ```
+
+2. add this pull secret to the ibm-infra-management-application-default service account, for example:
+
+    ```
+    oc secret link ibm-infra-management-application-default <pull-secret-name> --for=pull
+    ```
```
Comment (Fryguy) [inline managing_providers/_topics/embedded_workflows.md:297]: This should also use conrefs, and also be sentence cased.

```suggestion
2. Add this pull secret to the `{{ site.data.product.workflow_service_account }}` service account, for example:

    ```
    oc secret link {{ site.data.product.workflow_service_account }} <pull-secret-name> --for=pull
    ```
```

File: managing_providers/_topics/embedded_workflows.md [hunk 3]
```diff
@@ -280,6 +280,21 @@ Workflows must be authored in Amazon State Languages (ASL) format. As part of au
   }
 }
 ```
+#### Running an Embedded Workflow on the Openshift Container Platform (OCP)
```
Comment (Fryguy) [inline managing_providers/_topics/embedded_workflows.md:283]: Very minor, but probably good to add a blank line above this to separate it from the preview section.

```suggestion

#### Running an Embedded Workflow on the Openshift Container Platform (OCP) 
```

Comment (Fryguy) [reply managing_providers/_topics/embedded_workflows.md:283]: Also OpenShift should have a capital S

```suggestion
#### Running an Embedded Workflow on the OpenShift Container Platform (OCP) 
```

**Comment by Fryguy** [comment]
@agrare Aside from my comments, does the technical details of this look correct?

File: managing_providers/_topics/embedded_workflows.md (hunk 1 above)
Comment (Fryguy) [reply managing_providers/_topics/embedded_workflows.md:287]: It says "docker pull secret", but that doesn't sound right to me - I think it should just be "Create a pull secret".  Maybe I'm wrong though - @agrare Thoughts?

Comment (agrare) [reply managing_providers/_topics/embedded_workflows.md:287]: Technically you are creating a k8s `Secret` object and the purpose is up to the user, but I think `Pull Secret` or `Image Pull Secret` would be an accurate description.

File: managing_providers/_topics/embedded_workflows.md [hunk 4]
```diff
@@ -169,7 +169,7 @@ Workflows must be authored in Amazon State Languages (ASL) format. As part of au
 
    When you have the code for your task resource written, you need to bundle it into a docker image. You can bundle the code by creating a standard [Dockerfile](https://docs.docker.com/engine/reference/builder/) and building the image (https://docs.docker.com/engine/reference/commandline/build/). Then, you can push the image to a [registry](https://docs.docker.com/engine/reference/commandline/push/), which makes the image available to be used by {{ site.data.product.title_short }}. When you have pushed your images to an image registry, you can add the registry to {{ site.data.product.title_short }}. 
    
-  Pull secrets for containers are used differently between appliances and the OpenShift Container Platform (OCP).
+  Pull secrets for containers are used differently between appliances and the OpenShift Container Platform (OCP). These differences are outlined in the following sections.
 
    * On appliances, `podman` is used to execute the container
```
Comment (agrare) [inline managing_providers/_topics/embedded_workflows.md:174]: This line seems to be duplicated on the next line down

File: managing_providers/_topics/embedded_workflows.md [hunk 5]
```diff
@@ -187,11 +187,26 @@ Workflows must be authored in Amazon State Languages (ASL) format. As part of au
       
       You are recommended to use a docker.io [access token](https://docs.docker.com/security/for-developers/access-tokens/) so that the token does not expire.
 
-   * For OCP provide an image pull secret to a podified Kubernetes container, and then add it to a service  account
+   * Provide an image pull secret to a podified Kubernetes container, and then add it to a service account
```
Comment (agrare) [inline managing_providers/_topics/embedded_workflows.md:190]: Is this section needed now that we have `#### Running an Embedded Workflow on the Openshift Container Platform (OCP) ` below?

File: managing_providers/_topics/embedded_workflows.md [hunk 6]
```diff
@@ -187,11 +187,26 @@ Workflows must be authored in Amazon State Languages (ASL) format. As part of au
...
 
      In order to pull an image from a private registry you have to provide an `ImagePullSecret` to your containers, see [Pull an Image from a Private Registry](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/). {{ site.data.product.title_short }} uses a service account called {{ site.data.product.workflow_service_account }} to run containers for your workflows. You can add an `ImagePullSecret` to this service account by following [Add Image Pull Secrets to a service account](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#add-imagepullsecrets-to-a-service-account/).
 
-    
+#### Running an Embedded Workflow on the Openshift Container Platform (OCP)
```
Comment (agrare) [inline managing_providers/_topics/embedded_workflows.md:194]: Should there be a `#### Running an Embedded Workflow an Appliances` section above to match this?

File: managing_providers/_topics/embedded_workflows.md [hunk 7]
```diff
@@ -157,7 +157,7 @@ You can configure workflows for use in the following areas during service catalo
 
 You can create and use embedded workflows as needed to not only change parts of the provisioning process, but also to automate other operational tasks.
 
-Workflows must be authored in Amazon State Languages (ASL) format. As part of authoring a workflow, you (or your users) can build container images that are able to perform any tasks that are required in any language that you like. You can use these images during Task states in your workflows.
+Workflows must be authored in Amazon State Languages (ASL) format. As part of authoring a workflow, you (or your users) can build container images that are able to perform any tasks that are required in any language that you like. You can use these images during Task states in your workflows.
```
Comment (agrare) [inline managing_providers/_topics/embedded_workflows.md:160]: Please remove trailing whitespace

File: managing_providers/_topics/embedded_workflows.md [hunk 8]
```diff
@@ -167,26 +167,48 @@ Workflows must be authored in Amazon State Languages (ASL) format. As part of au
...
+   
+  Pull secrets for containers are used differently between appliances and the OpenShift Container Platform (OCP). These differences are outlined in the following sections.
 
-   * On appliances, `podman` is used to execute the container
+  #### Running an Embedded Workflow on Appliances
 
-     On appliances, `podman` is used to execute the container so use [podman login](https://docs.podman.io/en/stable/markdown/podman-login.1.html) as the `manageiq` user.
+   * On appliances, `podman` is used to execute the container
 
+     On appliances, `podman` is used to execute the container so use [podman login](https://docs.podman.io/en/stable/markdown/podman-login.1.html) as the `manageiq` user.
```
Comment (agrare) [inline managing_providers/_topics/embedded_workflows.md:178]:
```suggestion
   * On appliances, `podman` is used to execute the container so use [podman login](https://docs.podman.io/en/stable/markdown/podman-login.1.html) as the `manageiq` user. 
```

File: managing_providers/_topics/embedded_workflows.md [hunk 9]
```diff
@@ -280,7 +302,6 @@ Workflows must be authored in Amazon State Languages (ASL) format. As part of au
   }
 }
 ```
-
```
Comment (agrare) [inline managing_providers/_topics/embedded_workflows.md:283]: Was this removal intentional?  It doesn't seem related to the rest of these changes

**Comment by agrare** [comment]
I've made some changes in this branch so assigning @Fryguy for final review

**Review by Fryguy — APPROVED** [review]

**Comment by Fryguy** [comment]
Backported to `quinteros` in commit c1b05d16a73b1591ba9bdbc342803ca9e730c545.

```text
commit c1b05d16a73b1591ba9bdbc342803ca9e730c545
Author: Jason Frey <fryguy9@gmail.com>
Date:   Tue Nov 28 11:57:10 2023 -0500

    Merge pull request #1762 from aileenjackson/AJ_UpdateImagesEmbedWorkflow_latest
    
    Embedded Workflows on OCP docker custom pull secret
    
    (cherry picked from commit 649084fe659c44a62efbfb4affb4619bc3150320)
```
